src/f32-avgpool/gen/f32-avgpool-9p8x-minmax-rvv-c2v.c

```diff
@@ -172,90 +100,37 @@ void xnn_f32_avgpool_minmax_ukernel_9p8x__rvv_c2v(
         vfloat32m2_t sum0167a_f32v = __riscv_vfadd_vv_f32m2(sum01a_f32v, sum67_f32v, n);
         vfloat32m2_t sum_f32v = __riscv_vfadd_vv_f32m2(sum2345_f32v, sum0167a_f32v, n);
         __riscv_vse32_v_f32m2(b, sum_f32v, n); b += n;
-      } while (c != 0);
+      }
     }
 
     {
-      const float* i0 = input[0];
-      assert(i0 != NULL);
-      const float* i1 = input[1];
-      const float* i2 = input[2];
-      const float* i3 = input[3];
-      const float* i4 = input[4];
-      const float* i5 = input[5];
-      const float* i6 = input[6];
-      const float* i7 = input[7];
-      input = (const float**) ((uintptr_t) input + input_increment);
-      if (k < 2) {
-        i1 = zero;
-      }
-      assert(i1 != NULL);
-      if (k <= 2) {
-        i2 = zero;
-      }
-      assert(i2 != NULL);
-      if (k < 4) {
-        i3 = zero;
-      }
-      assert(i3 != NULL);
-      if (k <= 4) {
-        i4 = zero;
-      }
-      assert(i4 != NULL);
-      if (k < 6) {
-        i5 = zero;
-      }
-      assert(i5 != NULL);
-      if (k <= 6) {
-        i6 = zero;
-      }
-      assert(i6 != NULL);
-      if (k < 8) {
-        i7 = zero;
-      }
-      assert(i7 != NULL);
-      if XNN_UNPREDICTABLE(i0 != zero) {
-        i0 = (const float*) ((uintptr_t) i0 + input_offset);
-      }
-      if XNN_UNPREDICTABLE(i1 != zero) {
-        i1 = (const float*) ((uintptr_t) i1 + input_offset);
+      const float *i[8];
+      for (size_t kk = 0; kk < kernel_elements; ++kk) {
```

Comment:
Are these bounds right? I think that `kernel_elements` can be larger than 8 here, so that the bounds of array `i` will be exceeded.

Reply:
Thanks. Fixed it.